xc/domain.ml

```diff
@@ -230,15 +235,25 @@ let make ~xc ~xs vm_info uuid =
 				let ent = sprintf "%s/%s" dom_path dir in
 				t.Xst.mkdirperms ent roperm
 			) [ "cpu"; "memory" ];
+
+			let mksubdirs base dirs perms =
```

Comment:
I probably would have used `Filename.concat` and tried to be less clever with maps. Maybe I'm missing here something and you are doing some kind of multiplication. The `let () =` is only in there to try it in `utop`.


		let dompath = "/some/path"
		let rwperms = "r/w"
		let mkdirperms perms dir = Printf.printf "mkdir %s\n" dir

		let (//)  = Filename.concat 

		let mksubdirs perms base dirs =
			List.iter (fun dir -> mkdirperms perms (base // dir)) dirs


		let () = mksubdirs rwperms dompath
			 [ "feature"
			 ; "device"
			 ; "error"
			 ; "drivers"
			 ; "control"
			 ; "attr"
			 ; "data"
			 ; "messages"
			 ; "vm-data"
			 ; "hvmloader"
			 ; "rrd"
			 ; "device/vbd"
			 ; "device/vif"
			 ]